README
NAME
    Task::Tickit::Widget - Install the Tickit::Widget modules.

SYNOPSIS
     cpanm Task::Tickit::Widget

SEE ALSO
    Tickit::Widget
        Abstract base class for on-screen widgets.

    Tickit::Widget::Border
        Draw a fixed-size border around a widget.

    Tickit::Widget::Box
        Apply spacing and positioning to a widget.

    Tickit::Widget::Breadcrumb
        Render a breadcrumb trail.

    Tickit::Widget::Button
        A widget displaying a clickable button.

    Tickit::Widget::Calendar::MonthView
        ?

    Tickit::Widget::CheckButton
        A widget allowing a toggle true/false.

    Tickit::Widget::Choice
        A widget giving a choice from a list.

    Tickit::Widget::Decoration
        Do nothing, in a visually-appealing way.

    Tickit::Widget::Entry
        A widget for entering text.

    Tickit::Widget::Figlet
        Trivial wrapper around Text::FIGlet for banner rendering.

    Tickit::Widget::FileViewer
        Support for viewing files in Tickit.

    Tickit::Widget::Fill
        Fill an area with repeated text.

    Tickit::Widget::FloatBox
        Manage a collection of floating widgets.

    Tickit::Widget::Frame
        Draw a frame around another widget.

    Tickit::Widget::GridBox
        Lay out a set of child widgets in a grid.

    Tickit::Widget::HBox
        Distribute child widgets in a horizontal row.

    Tickit::Widget::HSplit
        An adjustable horizontal split between two widgets.

    Tickit::Widget::Layout::Desktop
        Provides a holder for "desktop-like" widget behaviour.

    Tickit::Widget::Layout::Relative
        Apply sizing to a group of Tickit widgets.

    Tickit::Widget::LinearBox
        Abstract base class for "HBox" and "VBox".

    Tickit::Widget::LinearSplit
        ?

    Tickit::Widget::LogAny
        Log message rendering.

    Tickit::Widget::Menu
        Display a menu of choices.

    Tickit::Widget::MenuBar
        Display a menu horizontally.

    Tickit::Widget::Menu::Item
        An item to display in a "Tickit::Widget::Menu".

    Tickit::Widget::Placegrid
        A placeholder grid display.

    Tickit::Widget::Progressbar
        Simple progressbar implementation for Tickit.

    Tickit::Widget::Progressbar::Horizontal
        Simple progressbar implementation for Tickit.

    Tickit::Widget::Progressbar::Vertical
        Simple progressbar implementation for Tickit.

    Tickit::Widget::RadioButton
        A widget allowing a selection from multiple options.

    Tickit::Widget::ScrollBox
        Allow a single child widget to be scrolled.

    Tickit::Widget::ScrollBox::Extent
        Represents the range of scrolling extent.

    Tickit::Widget::Scroller
        A widget displaying a scrollable collection of items.

    Tickit::Widget::Scroller::Item
        Interface for renderable scroller items.

    Tickit::Widget::Scroller::Item::RichText
        Static text with render attributes.

    Tickit::Widget::Scroller::Item::Text
        Add static text to a Scroller.

    Tickit::Widget::SegmentDisplay
        Show a single character like a segmented display.

    Tickit::Widget::SparkLine
        Minimal graph implementation for Tickit.

    Tickit::Widget::Spinner
        A widget displaying a small text animation.

    Tickit::Widget::Static
        A widget displaying static text.

    Tickit::Widget::Statusbar
        Provides a simple status bar implementation.

    Tickit::Widget::Statusbar::Clock
        A simple clock implementation.

    Tickit::Widget::Statusbar::CPU
        CPU usage.

    Tickit::Widget::Statusbar::Memory
        Memory usage.

    Tickit::Widget::Statusbar::WidgetList
        ?

    Tickit::Widget::Tabbed
        Provide tabbed window support.

    Tickit::Widget::Tabbed::Ribbon
        Base class for "Tickit::Widget::Tabbed" control ribbon.

    Tickit::Widget::Table
        Table widget with support for scrolling/paging.

    Tickit::Widget::Table::Paged
        Table widget with support for scrolling/paging.

    Tickit::Widget::Tree
        Tree widget implementation for Tickit.

    Tickit::Widget::VBox
        Distribute child widgets in a vertical column.

    Tickit::Widget::VHBox
        Distribute child widgets vertically or horizontally.

    Tickit::Widget::VSplit
        An adjustable vertical split between two widgets.

    Tickit::Widgets
        Load several Tickit::Widget classes at once.

REPOSITORY
    <https://github.com/tupinek/Task-Tickit-Widget>

AUTHOR
    Michal Špaček <mailto:skim@cpan.org>

    <http://skim.cz>

LICENSE AND COPYRIGHT
     © 2015 Michal Špaček
     Artistic License
     BSD 2-Clause License

VERSION
    0.06



Changes
0.06
 - Fix Makefile.PL fails with no '.' in @INC.
 - Update Tickit::Widgets to 0.23 version.

0.05 2016-02-23T14:23:27+01:00
 - Update Tickit::Widget::Choice to 0.02 version.
 - Update Tickit::Widget::ScrollBox to 0.07 version.
 - Update Tickit::Widget::Scroller to 0.22 version.
 - Update Tickit::Widgets to 0.22 version.

0.04 2015-10-30T08:48:31+01:00
 - Move author tests to xt/ directory.
 - Remove Tickit dependency, there are no widgets now.
 - Update Tickit::Widget::SegmentDisplay to 0.04 version.
 - Update Tickit::Widgets to 0.21 version.

0.03 2015-08-25T00:19:59+02:00
 - Add Tickit::Widget::Scroller 0.19 version.
 - Improve SEE ALSO section in doc.
 - Update Module::Install to 1.16 version.
 - Update Tickit to 0.53 version.
 - Update Tickit::Widget::FloatBox to 0.03 version.
 - Update Tickit::Widget::ScrollBox to 0.06 version.
 - Update Tickit::Widget::SparkLine to 0.105 version.
 - Update Tickit::Widget::Tabbed to 0.19 version.
 - Update Tickit::Widgets to 0.20 version.

0.02 2015-04-20T12:34:07+02:00
 - Add Tickit::Widget::Figlet 0.003 version.
 - Add Tickit::Widget::Term 0.002 version.
 - Update Module::Install to 1.15 version.
 - Update Tickit::Widget::Tabbed to 0.18 version.

0.01 2015-02-12T19:29:24+01:00
 - First version


Widget.pm
package Task::Tickit::Widget;

# Pragmas.
use strict;
use warnings;

# Version.
our $VERSION = 0.06;

1;

__END__

=pod

=encoding utf8

=head1 NAME

Task::Tickit::Widget - Install the Tickit::Widget modules.

=head1 SYNOPSIS

 cpanm Task::Tickit::Widget

=head1 SEE ALSO

=over

=item L<Tickit::Widget>

Abstract base class for on-screen widgets.

=item L<Tickit::Widget::Border>

Draw a fixed-size border around a widget.

=item L<Tickit::Widget::Box>

Apply spacing and positioning to a widget.

=item L<Tickit::Widget::Breadcrumb>

Render a breadcrumb trail.

=item L<Tickit::Widget::Button>

A widget displaying a clickable button.

=item L<Tickit::Widget::Calendar::MonthView>

?

=item L<Tickit::Widget::CheckButton>

A widget allowing a toggle true/false.

=item L<Tickit::Widget::Choice>

A widget giving a choice from a list.

=item L<Tickit::Widget::Decoration>

Do nothing, in a visually-appealing way.

=item L<Tickit::Widget::Entry>

A widget for entering text.

=item L<Tickit::Widget::Figlet>

Trivial wrapper around Text::FIGlet for banner rendering.

=item L<Tickit::Widget::FileViewer>

Support for viewing files in Tickit.

=item L<Tickit::Widget::Fill>

Fill an area with repeated text.

=item L<Tickit::Widget::FloatBox>

Manage a collection of floating widgets.

=item L<Tickit::Widget::Frame>

Draw a frame around another widget.

=item L<Tickit::Widget::GridBox>

Lay out a set of child widgets in a grid.

=item L<Tickit::Widget::HBox>

Distribute child widgets in a horizontal row.

=item L<Tickit::Widget::HSplit>

An adjustable horizontal split between two widgets.

=item L<Tickit::Widget::Layout::Desktop>

Provides a holder for "desktop-like" widget behaviour.

=item L<Tickit::Widget::Layout::Relative>

Apply sizing to a group of Tickit widgets.

=item L<Tickit::Widget::LinearBox>

Abstract base class for "HBox" and "VBox".

=item L<Tickit::Widget::LinearSplit>

?

=item L<Tickit::Widget::LogAny>

Log message rendering.

=item L<Tickit::Widget::Menu>

Display a menu of choices.

=item L<Tickit::Widget::MenuBar>

Display a menu horizontally.

=item L<Tickit::Widget::Menu::Item>

An item to display in a "Tickit::Widget::Menu".

=item L<Tickit::Widget::Placegrid>

A placeholder grid display.

=item L<Tickit::Widget::Progressbar>

Simple progressbar implementation for Tickit.

=item L<Tickit::Widget::Progressbar::Horizontal>

Simple progressbar implementation for Tickit.

=item L<Tickit::Widget::Progressbar::Vertical>

Simple progressbar implementation for Tickit.

=item L<Tickit::Widget::RadioButton>

A widget allowing a selection from multiple options.

=item L<Tickit::Widget::ScrollBox>

Allow a single child widget to be scrolled.

=item L<Tickit::Widget::ScrollBox::Extent>

Represents the range of scrolling extent.

=item L<Tickit::Widget::Scroller>

A widget displaying a scrollable collection of items.

=item L<Tickit::Widget::Scroller::Item>

Interface for renderable scroller items.

=item L<Tickit::Widget::Scroller::Item::RichText>

Static text with render attributes.

=item L<Tickit::Widget::Scroller::Item::Text>

Add static text to a Scroller.

=item L<Tickit::Widget::SegmentDisplay>

Show a single character like a segmented display.

=item L<Tickit::Widget::SparkLine>

Minimal graph implementation for Tickit.

=item L<Tickit::Widget::Spinner>

A widget displaying a small text animation.

=item L<Tickit::Widget::Static>

A widget displaying static text.

=item L<Tickit::Widget::Statusbar>

Provides a simple status bar implementation.

=item L<Tickit::Widget::Statusbar::Clock>

A simple clock implementation.

=item L<Tickit::Widget::Statusbar::CPU>

CPU usage.

=item L<Tickit::Widget::Statusbar::Memory>

Memory usage.

=item L<Tickit::Widget::Statusbar::WidgetList>

?

=item L<Tickit::Widget::Tabbed>

Provide tabbed window support.

=item L<Tickit::Widget::Tabbed::Ribbon>

Base class for "Tickit::Widget::Tabbed" control ribbon.

=item L<Tickit::Widget::Table>

Table widget with support for scrolling/paging.

=item L<Tickit::Widget::Table::Paged>

Table widget with support for scrolling/paging.

=item L<Tickit::Widget::Tree>

Tree widget implementation for Tickit.

=item L<Tickit::Widget::VBox>

Distribute child widgets in a vertical column.

=item L<Tickit::Widget::VHBox>

Distribute child widgets vertically or horizontally.

=item L<Tickit::Widget::VSplit>

An adjustable vertical split between two widgets.

=item L<Tickit::Widgets>

Load several Tickit::Widget classes at once.

=back

=head1 REPOSITORY

L<https://github.com/tupinek/Task-Tickit-Widget>

=head1 AUTHOR

Michal Špaček L<mailto:skim@cpan.org>

L<http://skim.cz>

=head1 LICENSE AND COPYRIGHT

 © 2015 Michal Špaček
 Artistic License
 BSD 2-Clause License

=head1 VERSION

0.06

=cut


Makefile.PL
# Pragmas.
use lib '.';
use strict;
use warnings;

# Modules.
use inc::Module::Install;

# Definition.
abstract 'Install the Tickit::Widget modules.';
author 'Michal Spacek <skim@cpan.org>';
author_requires 'English' => 0;
author_requires 'File::Object' => 0.08;
author_requires 'Test::More' => 0;
author_requires 'Test::NoWarnings' => 0;
author_requires 'Test::Pod' => 0;
author_requires 'Test::Pod::Coverage' => 0;
license 'bsd';
name 'Task-Tickit-Widget';
readme_from 'Widget.pm';
recursive_author_tests('xt');
requires 'Tickit::Widget::Breadcrumb' => 0.003;
requires 'Tickit::Widget::Calendars' => 0.001;
requires 'Tickit::Widget::Choice' => 0.02;
requires 'Tickit::Widget::Decoration' => 0.004;
requires 'Tickit::Widget::FileViewer' => 0.004;
requires 'Tickit::Widget::Figlet' => 0.003;
requires 'Tickit::Widget::FloatBox' => 0.03;
requires 'Tickit::Widget::Layout::Desktop' => 0.009;
requires 'Tickit::Widget::Layout::Relative' => 0.005;
requires 'Tickit::Widget::LogAny' => 0.004;
requires 'Tickit::Widget::Menu' => 0.09;
requires 'Tickit::Widget::Progressbar' => 0.101;
requires 'Tickit::Widget::ScrollBox' => 0.07;
requires 'Tickit::Widget::Scroller' => 0.22;
requires 'Tickit::Widget::SegmentDisplay' => 0.04;
requires 'Tickit::Widget::SparkLine' => 0.105;
requires 'Tickit::Widget::Statusbar' => 0.004;
requires 'Tickit::Widget::Tabbed' => 0.019;
requires 'Tickit::Widget::Table' => 0.214;
requires 'Tickit::Widget::Table::Paged' => 0.004;
requires 'Tickit::Widget::Term' => 0.002;
requires 'Tickit::Widget::Tree' => 0.114;
requires 'Tickit::Widget::VHBox' => 0.001;
requires 'Tickit::Widgets' => 0.23;
requires 'perl' => '5.6.0';
resources 'bugtracker' => 'https://rt.cpan.org/Public/Dist/Display.html'.
	'?Name=Task-Tickit-Widget';
resources 'homepage' => 'https://github.com/tupinek/Task-Tickit-Widget';
resources 'repository' => 'git://github.com/tupinek/Task-Tickit-Widget';
test_requires 'Test::More' => 0;
test_requires 'Test::NoWarnings' => 0;
tests_recursive;
version '0.06';

# Run.
WriteAll();
